CMakeLists.txt

```diff
@@ -82,6 +83,15 @@ if( EXIV2_BUILD_UNIT_TESTS )
     add_subdirectory ( unitTests )
 endif()
 
+if( EXIV2_BUILD_FUZZ_TESTS)
+    if (NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        message(FATAL_ERROR "You need to build with Clang for fuzzing to work")
+    endif()
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address,fuzzer-no-link -fsanitize-coverage=edge,indirect-calls")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address,fuzzer-no-link -fsanitize-coverage=edge,indirect-calls")
```

Comment:
Don't misuse CMAKE_C[XX]_FLAGS!

Also don't we have that in fuzz already?

Reply:
Hi @cryptomilk As far as I understand if we are fuzzing we need to compile the whole application with -fsanitize=address look at [Google](https://github.com/google/boringssl/blob/master/CMakeLists.txt) example. and this is why we add only -fsanitize=fuzzer when compiling the "main" of the fuzzer

Reply:
For that you define build types!

https://gitlab.com/libssh/libssh-mirror/blob/master/cmake/Modules/DefineCompilerFlags.cmake

Reply:
ok, will fix. why the exclamation mark:)

Reply:
You need to include the DefineCompilerFlags.cmake before the first project() call

Reply:
https://gitlab.com/libssh/libssh-mirror/blob/master/CMakeLists.txt#L11

Reply:
fixed. actually I noticed that there were already a flag for sanitizers (EXIV2_TEAM_USE_SANITIZERS). Thx for the note.

Reply:
EXIV2_TEAM_USE_SANITIZERS looks ugly. Could you also fix that correctly and add DefineCompilerFlags.cmake? Feel free to copy the file from libssh.

Reply:
what do you mean by adding DefineCompilerFlags.cmake? as far as I understand the file in this project is called cmake/compilerFlags.cmake. do you want to rename it?

Reply:
https://gitlab.com/libssh/libssh-mirror/blob/master/cmake/Modules/DefineCompilerFlags.cmake should be added and included correctly (before the project call). Then in cmake/compilerFlags.cmake the EXIV2_TEAM_USE_SANITIZERS stuff should be removed.

Reply:
ok and what about all the CMAKE_CXX_FLAGS that are set in [compilerFlags](https://github.com/Exiv2/exiv2/blob/master/cmake/compilerFlags.cmake) like BUILD_WITH_COVERAGE and others? also move them?

Reply:
@cryptomilk Push a fixed. let me know if it's ok now and if you want to move BUILD_WITH_COVERAGE to DefineCompilerFLags as well.

Reply:
That could be done with another PR. Thanks for cleaning this!